control-plane/agents/src/bin/core/node/service.rs

```diff
@@ -361,6 +361,11 @@ impl Service {
 
     /// Apply a drain label to the specified node. The reconciler will perform the drain.
     async fn drain(&self, id: NodeId, label: String) -> Result<Node, SvcError> {
+        // Don't allow draining if HA_ENABLED is false or undefined.
```

Comment:
does undefined wrap to false? Shouldn't undefined be defaulted as HA is enabled?

Reply:
It's upto us what we want undefined to unwrap to. I have made it false because if we let it be true then we have the same problem again i.e we might drain even if HA might be disabled.
For example for a case, someone removes the env variable from the deployment manually.

Reply:
Ok, does that mean we always have the helm variable for HA_ENABLED with default set to true? I was hinting if the variable HA_ENABLED itself was optional and default behavior was true...

Reply:
Yes, we will always have `HA_ENABLED=true` on the core-agent deployment as a default helm installation. 

Now if someone does a `helm install .... --set agents.ha.enabled=false` then `HA_ENABLED=false` will be set on the core-agent deployment.

We will get a undefined only when someone goes and edits the  core-agent deployment and removes the `HA_ENABLED` manually.

Reply:
Should it be the other way around, be enabled by default and disallowed when agents.ha.enabled=false ?
Then we don't need to keep env var unless it's disabled

Reply:
@tiagolobocastro Yes we can do that as well. What if its removed manually, we allow the drain assuming its all fine?

Reply:
Treat missing ha_enabled field same as ha_enabled = true

Reply:
Okay. Sure. I will make that change.